src/main/java/com/microsoft/sqlserver/jdbc/Parameter.java

```diff
@@ -528,9 +528,24 @@ private void setTypeDefinition(DTV dtv) {
                         if (userProvidesPrecision) {
                             param.typeDefinition = SSType.DECIMAL.toString() + "(" + valueLength + "," + scale + ")";
                         }
-                    } else
-                        param.typeDefinition = SSType.DECIMAL.toString() + "(" + SQLServerConnection.maxDecimalPrecision
-                                + "," + scale + ")";
+                    } else {
+                        BigDecimal bigDecimal = null;
+                        if (dtv.getJavaType() == JavaType.BIGDECIMAL &&
+                                null != (bigDecimal = (BigDecimal) dtv.getSetterValue())) {
+
+                            String plainValue = bigDecimal.abs().toPlainString();
+                            String[] plainValueArray = plainValue.split("\\.");
+                            int precision = plainValueArray.length == 2 ? plainValueArray[0].length() + plainValueArray[1].length() : plainValueArray[0].length();
```

Comment:
this var seems unnecessary it's only used once

Reply:
same with plainValue above

Reply:
and scale

Reply:
`precision`? How so?